### DISCUSSION TITLE
Making rules for folders if they're in a specific directory or contains a string?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [sxyazi/yazi#1759] [1. Q&A]
I looked at the tip for making folder specific rules if the path ends with a string: https://yazi-rs.github.io/docs/tips/#folder-rules, but not if it contains a string or inside a specific directory. In the example, `cx.active.current.cwd`is a URL type, but idk how to get that into a string or do what I want.

**Comment by AnirudhG07** [comment]
You can simply change the rule of code where it is changing for `ends_with` for "if it contains a string" ques. Check out [here](https://yazi-rs.github.io/docs/plugins/types/#shared.url) what all can be done with Url.
To see if it inside a specific directory, i believe the `parent()` will work

**Comment by sxyazi** [comment]
Url type has implemented the `__tostring` metamethod, so you can just `local str = tostring(url)`, https://yazi-rs.github.io/docs/plugins/types#shared.url

**Reply by YousufSSyed** [reply]
That works, I did `tostring(cwd):find("Downloads")`.

But how exactly do I use `parent()`? It says "Returns parent directory in string if any, otherwise nil", so if I wanted to see if the parent folder is named "SD", would I do `cwd.parent() == "SD"`?
